liquidity-mining/src/lib.rs

```diff
@@ -0,0 +1,1299 @@
+        DepositIdOverflow,
+
+        /// Deposit id is not valid.
+        InvalidDepositId,
+    }
+
+    /// Id sequencer for `GlobalPool` and `LiquidityPoolYieldFarm`.
+    #[pallet::storage]
+    #[pallet::getter(fn pool_id)]
+    pub type PoolIdSequencer<T: Config> = StorageValue<_, PoolId, ValueQuery>;
+
+    /// Sequencer for last 12 bytes of deposit id.
+    #[pallet::storage]
+    pub type DepositSequencer<T: Config> = StorageValue<_, DepositId, ValueQuery>;
+
+    /// Global pool details.
+    #[pallet::storage]
+    #[pallet::getter(fn global_pool)]
+    pub type GlobalPoolData<T: Config> = StorageMap<_, Twox64Concat, GlobalPoolId, GlobalPool<T>, OptionQuery>;
+
+    /// Liquidity pool yield farm details.
+    #[pallet::storage]
+    #[pallet::getter(fn liquidity_pool)]
+    pub type LiquidityPoolData<T: Config> = StorageDoubleMap<
+        _,
+        Twox64Concat,
+        GlobalPoolId,
+        Twox64Concat,
+        T::AmmPoolId,
+        LiquidityPoolYieldFarm<T>,
+        OptionQuery,
+    >;
+
+    /// Deposit details.
+    #[pallet::storage]
+    #[pallet::getter(fn deposit)]
+    pub type DepositData<T: Config> = StorageMap<_, Twox64Concat, DepositId, Deposit<T>, OptionQuery>;
+
+    /// `LiquidityPoolYieldFarm` metadata holding: `(existing deposits count, global pool id)`
+    #[pallet::storage]
+    #[pallet::getter(fn liq_pool_meta)]
+    pub type LiquidityPoolMetadata<T: Config> = StorageMap<_, Twox64Concat, PoolId, (u64, GlobalPoolId), OptionQuery>;
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {}
+}
+
+impl<T: Config> Pallet<T> {
+    /// Create new liquidity mining program with provided parameters.
+    ///
+    /// `owner` account have to have at least `total_rewards` balance. This funds will be
+    /// transferred from `owner` to farm account.
+    ///
+    /// Returns: `(global pool id, max reward per period)`
+    ///
+    /// Parameters:
+    /// - `total_rewards`: total rewards planned to distribute. This rewards will be
+    /// distributed between all liq. pools in liq. mining program.
+    /// - `planned_yielding_periods`: planned number of periods to distribute `total_rewards`.
+    /// WARN: THIS IS NOT HARD DEADLINE. Not all rewards have to be distributed in
+    /// `planned_yielding_periods`. Rewards are distributed based on the situation in the liq.
+    /// pools and can be distributed in a longer time frame but never in the shorter time frame.
+    /// - `blocks_per_period`:  number of blocks in a single period. Min. number of blocks per
+    /// period is 1.
+    /// - `incentivized_asset`: asset to be incentivized in AMM pools. All liq. pools added into
+    /// liq. mining program have to have `incentivized_asset` in their pair.
+    /// - `reward_currency`: payoff currency of rewards.
+    /// - `owner`: liq. mining farm owner.
+    /// - `yield_per_period`: percentage return on `reward_currency` of all pools
+    #[allow(clippy::too_many_arguments)]
+    #[transactional]
```

Comment:
I believe this will part of another transactional operation. 
```suggestion
    #[require_transactional]
```